plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/delete/IcebergPositionDeletePageSink.java

```diff
@@ -0,0 +1,162 @@
+import java.util.concurrent.CompletableFuture;
+
+import static io.airlift.slice.Slices.wrappedBuffer;
+import static io.trino.plugin.hive.util.ConfigurationUtils.toJobConf;
+import static java.util.Objects.requireNonNull;
+import static java.util.UUID.randomUUID;
+import static java.util.concurrent.CompletableFuture.completedFuture;
+import static org.apache.iceberg.io.DeleteSchemaUtil.pathPosSchema;
+
+public class IcebergPositionDeletePageSink
+        implements ConnectorPageSink
+{
+    private static final int MAX_PAGE_POSITIONS = 4096;
+    private static final Schema POSITION_DELETE_SCHEMA = pathPosSchema();
+
+    private final Optional<PartitionData> partition;
+    private final String outputPath;
+    private final JsonCodec<CommitTaskData> jsonCodec;
+    private final IcebergFileWriter writer;
+
+    private long writtenBytes;
+    private long systemMemoryUsage;
+    private long validationCpuNanos;
+    private boolean writtenData;
+
+    public IcebergPositionDeletePageSink(
+            PartitionSpec partitionSpec,
+            Optional<PartitionData> partition,
+            LocationProvider locationProvider,
+            IcebergFileWriterFactory fileWriterFactory,
+            HdfsEnvironment hdfsEnvironment,
+            HdfsContext hdfsContext,
+            JsonCodec<CommitTaskData> jsonCodec,
+            ConnectorSession session,
+            FileFormat fileFormat)
+    {
+        JobConf jobConf = toJobConf(hdfsEnvironment.getConfiguration(hdfsContext, new Path(locationProvider.newDataLocation("position-delete-file"))));
+        String fileName = fileFormat.addExtension("delete/position/" + randomUUID());
+        this.jsonCodec = requireNonNull(jsonCodec, "jsonCodec is null");
+        this.partition = requireNonNull(partition, "partition is null");
+        this.outputPath = partition.map(p -> locationProvider.newDataLocation(partitionSpec, p, fileName)).orElse(locationProvider.newDataLocation(fileName));
+        this.writer = fileWriterFactory.createFileWriter(new Path(outputPath), POSITION_DELETE_SCHEMA, jobConf, session, hdfsContext, fileFormat);
+    }
+
+    @Override
+    public long getCompletedBytes()
+    {
+        return writtenBytes;
+    }
+
+    @Override
+    public long getSystemMemoryUsage()
+    {
+        return systemMemoryUsage;
+    }
+
+    @Override
+    public long getValidationCpuNanos()
+    {
+        return validationCpuNanos;
+    }
+
+    @Override
+    public CompletableFuture<?> appendPage(Page page)
+    {
+        doAppend(page);
+        writtenData = true;
+        return NOT_BLOCKED;
+    }
+
+    @Override
+    public CompletableFuture<Collection<Slice>> finish()
+    {
+        Collection<Slice> commitTasks = new ArrayList<>();
+        if (writtenData) {
+            writer.commit();
+            CommitTaskData task = new CommitTaskData(
+                    outputPath,
+                    writer.getWrittenBytes(),
+                    new MetricsWrapper(writer.getMetrics()),
```

Comment:
Ideally columns of position delete should use full metrics mode. I will add that feature later.